cli/src/api/mod.rs

```diff
@@ -573,6 +589,66 @@ impl PhylumApi {
     }
 }
 
+/// Phylum group types.
+pub enum Group {
+    Legacy(String),
+    Org(OrgGroup),
+}
+
+impl Group {
+    /// Create a new group.
+    fn new(org: String, group: String) -> Self {
+        Self::Org(OrgGroup { org, name: group })
+    }
+
+    /// Create a group without organization.
+    fn new_legacy(group: String) -> Self {
+        Self::Legacy(group)
+    }
+
+    /// Create a group from an optional group name.
+    pub fn try_new<O, G>(org: Option<O>, group: Option<G>) -> Option<Self>
+    where
+        O: Into<String>,
+        G: Into<String>,
+    {
+        group.map(|group| match org {
+            Some(org) => Self::new(org.into(), group.into()),
+            None => Self::new_legacy(group.into()),
+        })
+    }
+
+    /// Get the group's organization.
+    pub fn org(&self) -> Option<&str> {
+        match self {
+            Self::Legacy(_) => None,
+            Self::Org(OrgGroup { org, .. }) => Some(org.as_str()),
+        }
+    }
+
+    /// Get the group's name.
+    pub fn name(&self) -> &str {
+        match self {
+            Self::Legacy(name) => name.as_str(),
+            Self::Org(OrgGroup { name, .. }) => name.as_str(),
+        }
+    }
+
+    /// Format the group as a single string.
+    pub fn combined_format(&self) -> Cow<'_, str> {
+        match self {
+            Self::Legacy(name) => Cow::Borrowed(name),
+            Self::Org(OrgGroup { org, name }) => Cow::Owned(format!("{org}/{name}")),
+        }
+    }
+}
+
+/// Group under an organization.
+pub struct OrgGroup {
+    org: String,
+    name: String,
+}
```

Comment:
Does this struct have any value? It seems like it's intended to be used as a private implementation detail of `Group`. Its members are always accessed through `Group`. It'd probably be better to just put the values into the variant unless we're planning to have a function that returns an `Option<OrgGroup>`.

Reply:
Moving structs outside of enums is **always** the better option. Anonymous structs in enums are strictly less powerful and should be avoided.

Reply:
It's less powerful, but also less verbose. We aren't using any of the additional capabilities.

Reply:
Not *yet*. Verbosity rarely matters much, but the additional capabilities are important and quite difficult to adopt at a later point in time.